storage/bbolt.go

```diff
@@ -101,11 +101,24 @@ func (b bboltStore) doTX(fn func(tx *bbolt.Tx) error, writable bool) error {
 		if err != nil {
 			return core.WrapError(ErrCommitFailed, err)
 		}
+		// Call AfterCommit functions
+		for _, opt := range opts {
+			if ac, ok := opt.(*afterCommit); ok {
+				ac.fn()
+			}
+		}
 	} else {
 		log.Logger().Warnf("Rolling back transaction application due to error: %s", appError)
 		err := dbTX.Rollback()
 		if err != nil {
 			log.Logger().Errorf("Could not rollback BBolt transaction: %s", err)
+		} else {
```

Comment:
should also be executed if rollback fails?

Reply:
Well, question is what happens with the data when rollback fails. Since it's also not committed, the TX will be abandoned and DB state unchanged?

Feels like the `AfterRollback` should actually be `BeforeRollback`?

Reply:
The hook is not for changing DB related data, but other data. Before or after rollback then doesn't really matter. In fact you can call it OnRollback :)

Reply:
For example, the in-memory IBLT/XOR tree have a change which needs to be rolled back when the write to disk fails. (By removing the tx or by reloading from disk)

Reply:
Fixed in lib.